lib/oli_web/common/select_timezone.ex

```diff
@@ -0,0 +1,27 @@
+defmodule OliWeb.Common.SelectTimezone do
+  use Phoenix.Component
+
+  import Phoenix.HTML.Form
+
+  alias Oli.Predefined
+  alias OliWeb.Router.Helpers, as: Routes
+
+  def render(assigns) do
+    ~H"""
+      <script>
+        function submitForm(){
+          const relativePath = window.location.pathname+window.location.search;
+          $('#hidden-redirect-to').val(relativePath);
+          $('#timezone-form').submit()
+        }
+      </script>
+
+      <%= form_for @conn, Routes.static_page_path(@conn, :update_timezone), [id: "timezone-form"], fn f -> %>
```

Comment:
I think we should display somewhere here what the system has determined the timezone to be. Perhaps instead of the default "Select Timezone" we indicate the timezone is determined from the browser and render the automatically determined timezone as the default. e.g. "Browser: America/New_York". What do you think?

<img width="335" alt="Screen Shot 2022-06-22 at 2 49 36 PM" src="https://user-images.githubusercontent.com/6248894/175115643-768679a9-2839-4eea-b7e4-76411bfa25f8.png">

Reply:
Regarding the default "Select Timezone" option that is causing the issue that you found, I forgot to remove that prompt from the options (good catch!), and it is not expected to return to the default behavior of determining my timezone automatically.
Currently this is how it works:
-  If the timezone is not set (`local_tz == nil` in the session), then the `timezone.ts` script is run to get the timezone from the browser and update it in the session.
- Once the timezone is set, the script is no longer run.
- If a user updates the timezone with this new feature, the `local_tz` is updated in the session so the original browser timezone is lost.
- If for some reason the timezone cannot be fetched from the browser, the system uses UTC by default.

Therefore, there is no way now to track if the timezone was obtained from the browser or not. We could store a map in the session instead like `%{timezone: "America/New_York", browser: true}` if we want to distinguish one timezone from the other. Anyway, is it necessary to make this distinction? I assumed once a user changed the default browser timezone to a new one, there was no need to go back.

Reply:
I see, I remember now we originally discussed this option would only be for the current session. I still think users will eventually expect this to behave like a preference, but that would require storing in the user record. So in order to restore auto selection, one just has to clear their session.

I think this is okay for now and if needed we can followup with a more user preference focused approach that stores the selection in the db.

Reply:
I agree this should behave like a preference, but wanted to keep it simple for this first version. I'll create a followup story to store it in the db for the user. Thanks!